### PR TITLE
rename factory to provider and clarify namespace

### PR DESCRIPTION
**Pull request by dyladan**
Now that https://github.com/open-telemetry/oteps/pull/76 has been merged, this is a follow up PR to address itss two major changes:

- rename factory to provider
- clarify the use of meter name as metric `namespace`

### CODE REVIEW

File: specification/api-metrics-user.md [hunk 1]
```diff
@@ -10,19 +10,19 @@ TODO: Add a table of contents.
...
+New `Meter` instances can be created via a `MeterProvider` and its `getMeter` method.
+`MeterProvider`s are generally expected to be used as singletons.
+Implementations SHOULD provide a single global default `MeterProvider`. The `getMeter` method expects two string arguments:
 
 - `name` (required): This name must identify the instrumentation library (also referred to as integration, e.g. `io.opentelemetry.contrib.mongodb`)
   and *not* the instrumented library.
-  In case an invalid name (null or empty string) is specified, a working default `Meter` implementation as a fallback is returned
+  In case an invalid name (null or empty string) is specified, a working default `Meter` implementation is returned as a fallback
   rather than returning null or throwing an exception.
-  A `MeterFactory` could also return a no-op `Meter` here if application owners configure the SDK to suppress telemetry produced by this library.
```
Comment (arminru) [inline specification/api-metrics-user.md]: Why is this deleted?
The (updated) OTEP says:
> Based on the name and version, a Provider could provide a no-op Tracer or *Meter* to specific instrumentation libraries

Comment (dyladan) [reply specification/api-metrics-user.md]: I removed it because the OTEP also specifically says something along the lines of "you could use this mechanism to disable tracers/meters, but that is outside the scope of this OTEP." I thought that meant the actual disable configuration was left to a future OTEP

**Comment by carlosalberto** [comment]
@dyladan Please rebase and push ;)
